views/site/404.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @var string $requestedUri
+ * @var \Yiisoft\Router\UrlGeneratorInterface $urlGenerator
+ */
+
+use Yiisoft\Html\Html;
+?>
+
+<div class="card shadow p-5 my-5 mx-5 bg-white rounded">
+    <div class="card-body text-center ">
+        <h1 class="card-title display-1 fw-bold">404</h1>
+        <p class="card-text">
+            <?php echo "The page "
+                . Html::span($requestedUri, ['class' => 'text-muted'])
```

Comment:
```php
. Html::span($urlMatcher->getCurrentUri()->getPath(), ['class' => 'text-muted'])`
```

Reply:
$requestedUri should be passed through `Html::encode()`.

Reply:
> $requestedUri should be passed through `Html::encode()`.

I changed my proposal accordingly 👍🏻